### ISSUE TITLE
Uncaught Error: Please define a provider before using any RpcObservable. when trying to run second Fether instance

### ISSUE THREAD

**Issue by ltfschoen** [paritytech/fether#333]
When I run a first instance of Fether with `yarn run start` and then run a second instance of Fether with `yarn run start` it opens the "Terms of Use" page and then crashes with error in the UI as shown below. However if I run the second instance of Fether with `yarn run electron` it displays the same terminal message but the UI doesn't crash and I can run that second instance:

```
Uncaught Error: Please define a provider before using any RpcObservable.
    at Object.push.../../node_modules/@parity/light.js/lib/api.js.exports.getApi (:3000/static/js/2.chunk.js:31559)
    at :3000/static/js/2.chunk.js:33014
    at SwitchMapSubscriber.chainName$ [as project] (:3000/static/js/2.chunk.js:32849)
    at SwitchMapSubscriber.push.../../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber._next (:3000/static/js/2.chunk.js:157042)
    at SwitchMapSubscriber.push.../../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (:3000/static/js/2.chunk.js:147956)
    at Array.<anonymous> (:3000/static/js/2.chunk.js:31353)
    at Object.emit (:3000/static/js/2.chunk.js:50283)
    at ComponentFromStream.componentWillMount (:3000/static/js/2.chunk.js:31383)
    at callComponentWillMount (:3000/static/js/2.chunk.js:120747)
    at mountClassInstance (:3000/static/js/2.chunk.js:120831)
    at updateClassComponent (:3000/static/js/2.chunk.js:122485)
    at beginWork (:3000/static/js/2.chunk.js:123185)
    at performUnitOfWork (:3000/static/js/2.chunk.js:125428)
    at workLoop (:3000/static/js/2.chunk.js:125468)
    at renderRoot (:3000/static/js/2.chunk.js:125505)
    at performWorkOnRoot (:3000/static/js/2.chunk.js:126153)
    at performWork (:3000/static/js/2.chunk.js:126075)
    at performSyncWork (:3000/static/js/2.chunk.js:126048)
    at requestWork (:3000/static/js/2.chunk.js:125938)
    at scheduleWork$1 (:3000/static/js/2.chunk.js:125801)
    at scheduleRootUpdate (:3000/static/js/2.chunk.js:126395)
    at updateContainerAtExpirationTime (:3000/static/js/2.chunk.js:126421)
    at updateContainer (:3000/static/js/2.chunk.js:126452)
    at ReactRoot.push.../../node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render (:3000/static/js/2.chunk.js:126751)
    at :3000/static/js/2.chunk.js:126905
    at unbatchedUpdates (:3000/static/js/2.chunk.js:126284)
    at legacyRenderSubtreeIntoContainer (:3000/static/js/2.chunk.js:126901)
    at Object.render (:3000/static/js/2.chunk.js:126965)
    at Module../src/index.js (:3000/static/js/main.chunk.js:7899)
    at __webpack_require__ (:3000/static/js/bundle.js:788)
    at fn (:3000/static/js/bundle.js:151)
    at Object.0 (:3000/static/js/main.chunk.js:9603)
    at __webpack_require__ (:3000/static/js/bundle.js:788)
    at checkDeferredModules (:3000/static/js/bundle.js:46)
    at Array.webpackJsonpCallback [as push] (:3000/static/js/bundle.js:33)
    at :3000/static/js/main.chunk.js:1
```

And in the terminal of the second instance is displays:

```
[start-electron]   [@parity/electron] external "@parity/electron" 42 bytes {main} [built]
[start-electron]   [electron] external "electron" 42 bytes {main} [built]
[start-electron]   [electron-webpack/out/electron-main-hmr/HmrClient] external "electron-webpack/out/electron-main-hmr/HmrClient" 42 bytes {main} [built]
[start-electron]   [source-map-support/source-map-support.js] external "source-map-support/source-map-support.js" 42 bytes {main} [built]
[start-electron]   [url] external "url" 42 bytes {main} [built]
[start-electron]   [0] multi ./node_modules/electron-webpack/out/electron-main-hmr/main-hmr ./src/main/index.js 40 bytes {main} [built]
[start-electron]       + 11 hidden modules
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   Debugger listening on ws://127.0.0.1:57518/0bcdbdbc-e409-49d1-b5e4-706deccae011
[start-electron]   For help see https://nodejs.org/en/docs/inspector
[start-electron] 
[start-electron] ┗ ----------------------------
[start-react   ] Something is already running on port 3000.
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:38.440Z] INFO (fether/19378 on lss-MacBook-Pro.local): Starting Parity Fether...
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:39.130Z] INFO (@parity/electron:main/19378 on lss-MacBook-Pro.local): Parity found on machine, can be run with "/Users/scon/Library/Application Support/Electron/parity".
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:39.169Z] INFO (@parity/electron:main/19378 on lss-MacBook-Pro.local): Another instance of parity is already running on http://127.0.0.1:8546, skip running local instance.
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:39.173Z] INFO (@parity/electron:main/19378 on lss-MacBook-Pro.local): Another instance of parity is already running on http://127.0.0.1:8546, skip running local instance.
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:44.521Z] INFO (@parity/electron:main/19378 on lss-MacBook-Pro.local): Requesting new token.
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:44.528Z] INFO (@parity/electron:main/19378 on lss-MacBook-Pro.local): Running "/Users/scon/Library/Application\ Support/Electron/parity signer new-token".
[start-electron] 
[start-electron] ┗ ----------------------------
[start-electron] ┏ Electron -------------------
[start-electron] 
[start-electron]   [2019-01-06T10:30:44.591Z] INFO (@parity/electron:main/19378 on lss-MacBook-Pro.local): Successfully extracted token.
[start-electron] 
[start-electron] ┗ ----------------------------
```

**Comment by ltfschoen** [comment]
Electron has a method for reporting this: https://electronjs.org/docs/api/app#event-second-instance

**Comment by Tbaut** [comment]
BTH I don't think we should bother too much with that, as you say yourself:
> Why would anyone want to run one more than one instance of Fether on the same host machine?

**Comment by amaurymartiny** [comment]
imo this issue is a deeper problem, not only happening when you run 2 instances, but randomly, due to some race condition.

The race is being raced between:
- `setApi()` in parityStore, which creates a `const api = new Api()` object, and does `setApi(api)`.
- all React components that require `setApi()` to be called before they are loaded.

Since #355, RequireHealthOverlay is only a Modal, so renders the children. There should be an additional mechanism that checks that the React components are only loaded after `setApi()` has been called.

Should be fixed by #412
